### DISCUSSION TITLE
Shared interfaces for BusinessListBase<T, C> and BusinessBindingListBase<T, C>

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#3534] [Questions]
I have a `ViewModelListBase` class which I want to work with either `BusinessListBase<T, C>` or `BusinessBindingListBase<T, C>`. I found it difficult to find the right common interfaces to do simple things like `AddNew()`. Currently, I have the following:

```cs
C   child       = default(C);

var bindingList = Model as IBindingList;

if (bindingList != null)
    child = (C)bindingList.AddNew();
else
{
    var observableBindingList = Model as IObservableBindingList;

    if (observableBindingList != null)
        child = (C)observableBindingList.AddNew();
}
```

`BusinessBindingListBase<T, C>` doesn't implement `IBusinessListBase<C>`. `IEditableCollection` wasn't much help. `BusinessListBase<T, C>` implements `IObservableBindingList` which sounded promising, but it's got nothing to do with `IBindingList` which `BusinessBindingListBase<T, C>` implements.

**Comment by rockfordlhotka** [comment]
If anyone comes up with a good solution I'd love to see it. Many years ago (2007?) I remember trying to reconcile the two collection types and never came up with something that would work in Windows Forms, WPF, and (at the time) Silverlight.
